src/aind_data_transfer/transformations/metadata_creation.py

```diff
@@ -65,3 +69,82 @@ def ephys_job_to_processing(
         )
 
         return processing_instance
+
+
+class SubjectMetadata:
+    """Class to handle the creation of the subject metadata file."""
+
+    output_file_name = "subject.json"
+
+    @staticmethod
+    def ephys_job_to_subject(
+        metadata_service_url: str,
+        subject_id: Optional[str] = None,
+        filepath: Optional[str] = None,
+        file_subject_regex: Optional[str] = (
+            "(ecephys|ephys)*_*(\\d+)_\\d{4}"
+        ),
+    ) -> Subject:
+        """
+
+        Parameters
+        ----------
+        metadata_service_url : str
+        subject_id : Optional[str]
+          The subject id. If not provided, this method will try to parse it
+          from the filepath. Default is None
+        filepath : Optional[str]
+          If the subject_id is None, then this method will try to parse the
+          subject id from the filepath. Default is None.
+        file_subject_regex : Optional[str]
+          Regex pattern that will be used if the subject_id is None and the
+          filepath is not None.
+
+        Returns
+        -------
+        Subject
+          The subject information retrieved from aind-metadata-service.
+
+        """
+
+        def try_parse_data(d) -> Subject:
```

Comment:
I'm not sure we want to try to do validation here. The service is already validating at a particular version.  If you validate here you will have to always keep everything in sync.

Reply:
That makes sense. I can create a json file based on the response directly.